types/pluck.d.ts

```diff
@@ -1,10 +1,10 @@
 import { Placeholder } from 'ramda';
 
 export function pluck<K extends PropertyKey>(prop: K extends Placeholder ? never : K): {
-  <U extends readonly unknown[] | Record<K, any>>(obj: Record<PropertyKey, U>): U extends readonly (infer T)[] ? T[] : U extends Record<K, infer T> ? T[] : never;
+  <U extends Record<K, any>, IK extends string>(obj: Record<IK, U>): { [KK in keyof typeof obj]: U[K] };
   <U extends readonly unknown[] | Record<K, any>>(list: U[]): U extends readonly (infer T)[] ? T[] : U extends Record<K, infer T> ? T[] : never;
 };
-export function pluck<U>(__: Placeholder, obj: Record<PropertyKey, U>): <K extends keyof U>(prop: K) => Array<U[K]>;
-export function pluck<U>(__: Placeholder, list: readonly U[]): <K extends keyof U>(prop: K) => Array<U[K]>;
-export function pluck<K extends keyof U, U>(prop: K, obj: Record<PropertyKey, U>): Array<U[K]>;
+export function pluck<K extends keyof U, U extends Record<any, any>, IK extends keyof any>(prop: K, record: Record<IK, U>): { [KK in keyof typeof record]: U[K] };
 export function pluck<K extends keyof U, U>(prop: K, list: readonly U[]): Array<U[K]>;
+export function pluck<U extends Record<any, any>, IK extends keyof any>(__: Placeholder, record: Record<IK, U>): <K extends keyof U>(prop: K) => { [KK in keyof typeof record]: U[K] };
+export function pluck<U>(__: Placeholder, list: readonly U[]): <K extends keyof U>(prop: K) => Array<U[K]>;
```

Comment:
Typescript always selects the last defined overload when you pass an overloaded function like a an argument to another function. That means if you called `flip(pluck)`, `flip` would get the `function pluck<U>(__: Placeholder, list: readonly U[]): <K extends keyof U>(prop: K) => Array<U[K]>` type definition and return `<U>(list: readonly U[], __: Placeholder) => <K extends keyof U>(prop: K) => Array<U[K]>`, and that's not what we want.

The order was already correct, please re-order your changes to match the original order